clientv3/client.go

```diff
@@ -334,43 +279,30 @@ func (c *Client) dial(target string, creds grpccredentials.TransportCredentials,
 		defer cancel() // TODO: Is this right for cases where grpc.WithBlock() is not set on the dial options?
 	}
 
+	initialEndpoints := strings.Join(c.cfg.Endpoints, ";")
+	target := fmt.Sprintf("%s://%p/#initially=[%s]", resolver.Schema, c, initialEndpoints)
```

Comment:
Please remember that this code was affected by bug https://github.com/etcd-io/etcd/issues/13192

We should consider backporting https://github.com/etcd-io/etcd/issues/13192. The PR is pretty big as it implemented both integration and e2e test. E2e tests required a lot of work on the framework so I would skip it.

So we should backport at least:
* integration test https://github.com/etcd-io/etcd/pull/13375/commits/58d2b12a5016956fa9b451c77c3779f7c03f13fd
* fix https://github.com/etcd-io/etcd/pull/13375/commits/79f9a4557461783f467679e869bd143228d9d406

Reply:
It (https://github.com/etcd-io/etcd/pull/13359) has already been tracked in the original huge PR https://github.com/etcd-io/etcd/pull/16827. 

I requested @chaochn47 to backport the PR step by step.

Reply:
Also added a link to this comment, thx anyway.

Reply:
Thanks all! Regarding how to backport the test, we can discuss separately. I feel backporting e2e test is easier than integration test...